config/manager/kustomization.yaml

```diff
@@ -4,5 +4,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: controller
-  newTag: latest
+  newName: remontada007/checkpoint-restore-operator
```

Comment:
Do we need to use a container registry here?

Reply:
I think we should, currently user need to push operator image to the registry and then they can use that by `make deploy` so it would be good to have a registry so that user don't have to push the image to container registry.

Reply:
@adrianreber Would it make sense to create a container image in quay.io that builds automatically from the GitHub repository?

Reply:
I am also not totally convinced about this. 

If we have to have to have the image in a registry then let's use the github registry: ghcr.io

That way we could easily push during one of our GitHub Actions runs.

Reply:
@rst0git could you check once why our operator can't recognize the checkpoints in `/var/lib/kubelet/checkpoints` directory which is manually created. Here in the log its showing [checkpoints](https://github.com/Parthiba-Hazra/checkpoint-restore-operator/actions/runs/9861467129/job/27229963037#step:12:274) are present but operator could not able to [find](https://github.com/Parthiba-Hazra/checkpoint-restore-operator/actions/runs/9861467129/job/27229963037#step:12:299) them.

Reply:
Could it possibly be due to the naming of the checkpoints? The format uses the pod name and namespace name, which might be used to identify them.

Reply:
> @rst0git could you check once why our operator can't recognize the checkpoints in `/var/lib/kubelet/checkpoints` directory which is manually created. Here in the log its showing [checkpoints](https://github.com/Parthiba-Hazra/checkpoint-restore-operator/actions/runs/9861467129/job/27229963037#step:12:274) are present but operator could not able to [find](https://github.com/Parthiba-Hazra/checkpoint-restore-operator/actions/runs/9861467129/job/27229963037#step:12:299) them.

We need to split the changes into multiple commits, and run the tests after each commit to confirm that the operator can find checkpoints. This would make it much easier to identify the problem.

Reply:
> If we have to have to have the image in a registry then let's use the github registry: ghcr.io

I haven't used ghcr.io, but I think it may not be included in the list of default registries for container runtimes. It looks like most projects that provide "official container images" use DockerHub.

Reply:
@rst0git @adrianreber any conclusion here ?

Reply:
I created the following repository: https://quay.io/repository/criu/checkpoint-restore-operator
It is currently configured to automatically build an image from my github fork as I don't have permissions in this repository.
@adrianreber I also sent you an invite in quay.io

Reply:
@Parthiba-Hazra For testing in CI we can use a local registry:
https://docs.docker.com/build/ci/github-actions/local-registry/

Reply:
@rst0git Can you once look into this [action run](https://github.com/Parthiba-Hazra/checkpoint-restore-operator/actions/runs/10237803102/job/28321553318). I am using local registry but the kind can't pull the image in cluster. I tried connecting kind network to the registry but it looks something I am missing or unaware of.

Reply:
For now can we load the image to cluster using `kind load` and later I can work on connecting the registry to kind cluster on different pr?

Reply:
This is an interesting problem. Have tried the steps in the following example?
https://kind.sigs.k8s.io/docs/user/local-registry/#create-a-cluster-and-registry

Reply:
@rst0git https://github.com/Parthiba-Hazra/checkpoint-restore-operator/tree/ci-test

Reply:
It looks like the registry name is not set correctly:
```
2024-08-05T18:38:57.9830105Z env:
2024-08-05T18:38:57.9830366Z   CHECKPOINT_DIR: /var/lib/kubelet/checkpoints
2024-08-05T18:38:57.9830735Z   REGISTRY_PORT: 5000
2024-08-05T18:38:57.9831327Z   REGISTRY_NAME: 6ebd9f2bd7f0612642d3939c22951f18a9f8a7ef3d0a4a523b7eeb7f40ef240f
2024-08-05T18:38:57.9831920Z ##[endgroup]
2024-08-05T18:38:58.1077140Z [host."http://6ebd9f2bd7f0612642d3939c22951f18a9f8a7ef3d0a4a523b7eeb7f40ef240f:5000"]
2024-08-05T18:38:58.1239992Z template parsing error: template: :1:23: executing "" at <.NetworkSettings.Networks.kind>: map has no entry for key "NetworkSettings"
```

```
Failed to pull image "localhost:5000/checkpoint-restore-operator:ci": failed to pull and unpack image "localhost:5000/checkpoint-restore-operator:ci": failed to resolve reference "localhost:5000/checkpoint-restore-operator:ci": failed to do request: Head "http://6ebd9f2bd7f0612642d3939c22951f18a9f8a7ef3d0a4a523b7eeb7f40ef240f:5000/v2/checkpoint-restore-operator/manifests/ci?ns=localhost%3A5000": dial tcp: lookup 6ebd9f2bd7f0612642d3939c22951f18a9f8a7ef3d0a4a523b7eeb7f40ef240f: no such host
```

Reply:
It succeed :)  https://github.com/Parthiba-Hazra/checkpoint-restore-operator/actions/runs/10261726973 
but again it was failing in local run - 
```
Failed to pull image "localhost:5000/checkpoint-restore-operator:ci": failed to pull and unpack image "localhost:5000/checkpoint-restore-operator:ci": failed to resolve reference "localhost:5000/checkpoint-restore-operator:ci": failed to do request: Head "http://act-Build-and-Deploy-build-and-deploy-6182acab52e9c07604d5a4bea-9540cd62341ed14f1812a6948a4a81de498225f0235e667589bb3dcb5740c36a:5000/v2/checkpoint-restore-operator/manifests/ci?ns=localhost%3A5000": dial tcp: lookup act-Build-and-Deploy-build-and-deploy-6182acab52e9c07604d5a4bea-9540cd62341ed14f1812a6948a4a81de498225f0235e667589bb3dcb5740c36a: no such host
```